### PR TITLE
rewrite assign executor jdbc methods

### PR DESCRIPTION
**Pull request by kunkun-tang**
Still, this PR is a follow-up of #1345. Last but not least, I'm moving assignExecutor and UnAssignExecutor to a new DAO file.

The reason I don't create a new test file:
* Not usre how much in-memory h2 database consume resources. We could use only one test class (in-memory h2) to do testing.

### CODE REVIEW

File: azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java [hunk 1]
```diff
@@ -176,6 +182,47 @@ public void testFetchQueuedFlows() throws Exception {
...
+  }
+
+  /* Test exception when assigning a non-existent executor to a flow */
+  @Test
+  public void testAssignExecutorInvalidExecutor() throws Exception {
+    final ExecutableFlow flow = TestUtils.createExecutableFlow("exectest1", "exec1");
+    this.executionFlowDao.uploadExecutableFlow(flow);
+
+    assertThatThrownBy(
+        () -> this.assignExecutor.assignExecutor(flow.getExecutionId(), 1))
```
Comment (jamiesjc) [inline azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java]: Should it be assignExecutor(1, flow.getExecutionId())? Also how do you know 1 is a non-existent executor?

File: azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java [hunk 2]
```diff
@@ -176,6 +182,47 @@ public void testFetchQueuedFlows() throws Exception {
...
+
+  /* Test exception when assigning an executor to a non-existent flow execution */
+  @Test
+  public void testAssignExecutorInvalidExecution() throws Exception{
+    final String host = "localhost";
+    final int port = 12345;
+    final Executor executor = this.executorDao.addExecutor(host, port);
+
+    assertThatThrownBy(
+        () -> this.assignExecutor.assignExecutor(2, executor.getId()))
```
Comment (jamiesjc) [inline azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java]: Similar to above comment. Need to switch the two arguments.

File: azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java (hunk 1 above)
Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java]: Great Catch! fixed and added a comment.

File: azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java (hunk 2 above)
Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java]: Great Catch! fixed and added a comment.

**Pull request merged**